### ISSUE TITLE
大能源比赛

### ISSUE THREAD

**Issue by MiaoRain** [MiaoRain/lund#26]
![image](https://user-images.githubusercontent.com/47746159/79635190-c84c0680-816f-11ea-88d5-7c18596414f3.png)
![image](https://user-images.githubusercontent.com/47746159/79635251-35f83280-8170-11ea-8aa1-3d6af2064c86.png)
![image](https://user-images.githubusercontent.com/47746159/79635262-47d9d580-8170-11ea-955d-f72014a774db.png)
![image](https://user-images.githubusercontent.com/47746159/79635279-56c08800-8170-11ea-8a1e-982727faaa05.png)
![image](https://user-images.githubusercontent.com/47746159/79635359-b28b1100-8170-11ea-99f9-3d68ff11813a.png)

**Comment by MiaoRain** [comment]
**EDA**
https://www.kaggle.com/miaorain/ashrae-start-here-a-gentle-introduction
https://www.kaggle.com/nroman/eda-for-ashrae
interpolate

**Comment by MiaoRain** [comment]
**LightGBM 调参方法（具体操作）**
https://www.cnblogs.com/bjwu/p/9307344.html
**LightGBM 如何调参**
https://www.jianshu.com/p/b4ac0596e5ef

![image](https://user-images.githubusercontent.com/47746159/82343858-a3a7b080-99f3-11ea-8e3a-d6117afec7cd.png)
![image](https://user-images.githubusercontent.com/47746159/82343964-c4700600-99f3-11ea-8aa6-be06cb129e17.png)
![image](https://user-images.githubusercontent.com/47746159/82345017-0ea5b700-99f5-11ea-943a-71b02a71ee16.png)

params = {
    "objective": "regression",
    "boosting": "gbdt",
    "num_leaves": 1280,
    "learning_rate": 0.05,
    "feature_fraction": 0.85,
    "reg_lambda": 2,
    "metric": "rmse",
}
1. 首先选择较高的**学习率，大概0.1附近**，这样是为了加快收敛的速度。这对于调参是很有必要的。
    "objective": "regression",
    "boosting": "gbdt",
   "learning_rate": 0.05,
2. 对**决策树**基本参数调参
****"num_leaves"**: 1280,**
这是提高精确度的最重要的参数。
max_depth ：设置树深度，深度越大可能过拟合
num_leaves：因为 LightGBM 使用的是 leaf-wise 的算法，因此在调节树的复杂程度时，使用的是 num_leaves 而不是 max_depth。大致换算关系：num_leaves = 2^(max_depth)，但是它的值的设置应该小于 2^(max_depth)，否则可能会导致过拟合。

3. **正则化**参数调参
这两个参数都是为了**降低过拟合**的。
**feature_fraction**参数来进行特征的子抽样。这个参数可以用来防止过拟合及提高训练速度。
bagging_fraction+bagging_freq参数必须同时设置，bagging_fraction相当于subsample样本采样，可以使bagging更快的运行，同时也可以降拟合。bagging_freq默认0，表示bagging的频率，0意味着没有使用bagging，k意味着每k轮迭代进行一次bagging。

4. 最后**降低学习率**，这里是为了最后提高准确率

**Comment by MiaoRain** [comment]
**LightGBM + XGBoost + Catboost**
https://www.kaggle.com/samratp/lightgbm-xgboost-catboost

cb_model = CatBoostRegressor(iterations=500,
                             learning_rate=0.05,
                             depth=10,
                             eval_metric='RMSE',
                             random_seed = 42,
                             bagging_temperature = 0.2,
                             od_type='Iter',
                             metric_period = 50,
                             od_wait=20)

**Comment by MiaoRain** [comment]
cb_model = CatBoostRegressor(iterations=500,
                             learning_rate=0.05,
                             depth=10,
                             eval_metric='RMSE',
                             random_seed = 42,
                             bagging_temperature = 0.2,
                             od_type='Iter',
                             metric_period = 50,
                             od_wait=20)

**Comment by MiaoRain** [comment]
**Pandas详解十一之Fillna填充缺失数据**
https://blog.csdn.net/weixin_38168620/article/details/79596819
**python numpy中setdiff1d的用法**
https://blog.csdn.net/qq_36556893/article/details/89916399
**[3] DataFrame 数据合并，连接（merge,join,concat)**
https://blog.csdn.net/zutsoft/article/details/51498026

**Comment by MiaoRain** [comment]
![image](https://user-images.githubusercontent.com/47746159/82676149-33449d80-9c46-11ea-82dc-15fc728f4d63.png)

**Comment by MiaoRain** [comment]
**https://www.bilibili.com/video/BV1ub411x7aR?from=search&seid=10638790195027627815**
![image](https://user-images.githubusercontent.com/47746159/82694219-2d10ea00-9c63-11ea-8b41-3e48329af335.png)

![image](https://user-images.githubusercontent.com/47746159/82694854-4cf4dd80-9c64-11ea-8be5-965f0a68b03f.png)

![image](https://user-images.githubusercontent.com/47746159/82694753-1f0f9900-9c64-11ea-9691-b9a391082a02.png)
